src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -1111,6 +1140,51 @@ void FancyZones::OnEditorExitEvent() noexcept
     JSONHelpers::FancyZonesDataInstance().SaveFancyZonesData();
 }
 
+std::vector<HMONITOR> FancyZones::GetMonitorsSorted()
+{
+    std::shared_lock readLock(m_lock);
+
+    std::vector<std::pair<HMONITOR, RECT>> monitorInfo;
+    for (const auto& [monitor, window] : m_zoneWindowMap)
+    {
+        if (window->ActiveZoneSet() != nullptr)
+        {
+            MONITORINFOEX mi;
+            mi.cbSize = sizeof(mi);
+            GetMonitorInfo(monitor, &mi);
+            monitorInfo.push_back({ monitor, mi.rcMonitor });
+        }
+    }
+
+    auto isGreater = [](const std::pair<HMONITOR, RECT>& lhs, const std::pair<HMONITOR, RECT>& rhs) {
+        if (lhs.second.left > rhs.second.left)

```

Comment:
maybe `std::tie`, like `return std::tie(lhs.second.left,lhs.second.top) > std::tie(rhs.second.left,rhs.second.top);`?

Reply:
Well yeah, but we have better improvement on this from @ivan100sic, more generic and robust.